### PR TITLE
Show names of erroneous files in case of parsing errors while executing table functions

### LINKED ISSUE #34198: `file` table function should print the name of erroneous file in case of error
**Describe the issue**

```
$ echo "Hello,World" > test1.csv
$ echo "Error" > test2.csv

$ clickhouse-local --query "SELECT * FROM file('test*.csv', CSV, 'a String, b String')"

Hello   World
Code: 27. DB::Exception: Cannot parse input: expected ',' before: '\n': 
Row 1:
Column 0,   name: a, type: String, parsed text: "Error"
ERROR: Line feed found where delimiter (,) is expected. It's like your file has less columns than expected.
And if your file has the right number of columns, maybe it has unescaped quotes in values.

: While executing ParallelParsingBlockInputFormat: While executing File. (CANNOT_PARSE_INPUT_ASSERTION_FAILED)
```

There is no information that the error is related to `test2.csv`.

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Show names of erroneous files in case of parsing errors while executing table functions `file`, `s3` and `url`.

Resolves #34198.

### CODE REVIEW

File: src/IO/ReadBufferFromS3.h [hunk 1]
```diff
@@ -25,7 +26,7 @@ namespace DB
 /**
  * Perform S3 HTTP GET request and provide response to read.
  */
-class ReadBufferFromS3 : public SeekableReadBufferWithSize
+class ReadBufferFromS3 : public SeekableReadBufferWithSize, public WithFileName
```
Comment (kssenii) [inline src/IO/ReadBufferFromS3.h:29]: also `ReadBufferFromHDFS` ?

**Review by kssenii — APPROVED** [review]
LGTM

File: src/IO/ReadBufferFromS3.h (hunk 1 above)
Comment (CurtizJ) [reply src/IO/ReadBufferFromS3.h:29]: Yes, and maybe `AzureBlobStorage`.

**Comment by kssenii** [comment]
> Integration tests (release, actions) [1/2] — fail: 105 

docker issue

> Stateless tests flaky check (address, actions) — fail: 75, passed: 125

failed because the same test was run in parallel to itself (creates files with the same name as test name), possible only in flaky check.

**Pull request merged**